src/components/MTls/MeshMTLSStatus.tsx

```diff
@@ -48,25 +47,17 @@ const statusDescriptors = new Map<string, StatusDescriptor>([
 
 class MeshMTLSStatus extends React.Component<Props, State> {
   componentDidMount() {
-    const intervalId = this.setIntervalId(this.props.refreshInterval);
+    this.fetchStatus();
     this.setState({
-      intervalId: intervalId,
-      refreshInterval: this.props.refreshInterval
+      lastRefreshAt: Date.now()
     });
   }
 
-  componentWillUnmount() {
-    this.clearIntervalId();
-  }
-
   componentDidUpdate() {
-    if (this.props.refreshInterval !== this.state.refreshInterval) {
-      this.clearIntervalId();
-
-      const intervalId = this.setIntervalId(this.props.refreshInterval);
+    if (this.props.lastRefreshAt > this.state.lastRefreshAt) {
```

Comment:
if you get rid of the component state this can just be:

```
 componentDidUpdate(prevProps: Props, props: Props) {
   if (this.props.lastRefreshAt !== this.prevProps.lastRefreshAt {
      this.fetchStatus();
  }
}
```